src/gui/guiConfirmRegistration.cpp

```diff
@@ -0,0 +1,243 @@
+WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+
+#include "guiConfirmRegistration.h"
+#include "client.h"
+#include <IGUICheckBox.h>
+#include <IGUIEditBox.h>
+#include <IGUIButton.h>
+#include <IGUIStaticText.h>
+#include <IGUIFont.h>
+
+#include "gettext.h"
+
+// Continuing from guiPasswordChange.cpp
+const int ID_confirmPassword = 262;
+const int ID_confirm = 263;
+const int ID_message = 264;
+const int ID_cancel = 265;
+
+GUIConfirmRegistration::GUIConfirmRegistration(gui::IGUIEnvironment *env,
+		gui::IGUIElement *parent, s32 id, IMenuManager *menumgr, Client *client,
+		const std::string &playername, const std::string &password,
+		const std::string &address, bool *aborted) :
+		GUIModalMenu(env, parent, id, menumgr),
+		m_client(client), m_playername(playername), m_password(password),
+		m_address(address), m_aborted(aborted)
+{
+}
+
+GUIConfirmRegistration::~GUIConfirmRegistration()
+{
+	removeChildren();
+}
+
+void GUIConfirmRegistration::removeChildren()
+{
+	const core::list<gui::IGUIElement *> &children = getChildren();
+	core::list<gui::IGUIElement *> children_copy;
+	for (gui::IGUIElement *i : children) {
+		children_copy.push_back(i);
+	}
+
+	for (gui::IGUIElement *i : children_copy) {
+		i->remove();
+	}
+}
+void GUIConfirmRegistration::regenerateGui(v2u32 screensize)
+{
+	/*
+		save current input
+	*/
+	acceptInput();
+
+	/*
+		Remove stuff
+	*/
+	removeChildren();
+
+	/*
+		Calculate new sizes and positions
+	*/
+	core::rect<s32> rect(screensize.X / 2 - 600 / 2, screensize.Y / 2 - 300 / 2,
+			screensize.X / 2 + 600 / 2, screensize.Y / 2 + 300 / 2);
+
+	DesiredRect = rect;
+	recalculateAbsolutePosition(false);
+
+	v2s32 size = rect.getSize();
+	v2s32 topleft_client(0, 0);
+
+	const wchar_t *text;
+
+	/*
+		Add stuff
+	*/
+	s32 ypos = 30;
+	{
+		std::string address = m_address;
+		if (address == "") {
+			address = "localhost";
+		}
+		core::rect<s32> rect(0, 0, 540, 90);
+		rect += topleft_client + v2s32(30, ypos);
+		static const std::string info_text_template = strgettext(
+				"You are about to join the server (%s) with name "
+				"and password, but no player account is registered "
+				"on the name \"%s\" yet. If you proceed, a new "
+				"password-protected player account with this name "
+				"will be created on this server.\nIf you want to "
+				"create an account and join this server, please "
+				"enter the previously entered password once again "
+				"to confirm account creation:");
+		char info_text_buf[600];
+		snprintf(info_text_buf, ARRLEN(info_text_buf), info_text_template.c_str(),
```

Comment:
IIRC, doesn't gettext offer a pattern matching function for filling in variables in translated strings?